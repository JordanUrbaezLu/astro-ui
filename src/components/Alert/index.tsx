/* istanbul ignore file */
export * from "./Alert";
