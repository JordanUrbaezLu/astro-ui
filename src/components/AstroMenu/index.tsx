export * from "./AstroMenu";
export * from "./AstroMenuItem";
