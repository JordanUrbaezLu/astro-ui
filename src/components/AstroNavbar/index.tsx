export * from "./AstroNavbar";
export * from "./AstroNavbarItem";
