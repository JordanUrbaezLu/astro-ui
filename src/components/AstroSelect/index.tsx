export * from "./AstroSelect";
export * from "./AstroSelectItem";
