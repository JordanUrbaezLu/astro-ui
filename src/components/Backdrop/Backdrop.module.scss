.backdrop {
  background-color: rgb(0 0 0 / 30%);
  bottom: 0;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 999;

  &.noBackground {
    background-color: transparent;
  }
}
