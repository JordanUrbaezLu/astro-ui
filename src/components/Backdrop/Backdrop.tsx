import classNames from "classnames";
import * as React from "react";
import styles from "./Backdrop.module.scss";

export interface BackdropProps
  extends React.ComponentPropsWithoutRef<"div"> {
  /**
   * The content for the Backdrop
   */
  children?: React.ReactNode;
  /**
   * If the Backdrop shows the background
   *
   * @default true
   */
  showBackground?: boolean;
}

/**
 * @public
 *
 * @description
 *
 * The Backdrop component will add a dimmed layer over the page
 * and is used to empahisize a particular element.
 */
export const Backdrop: React.FC<BackdropProps> = (props) => {
  const {
    className,
    children,
    showBackground = true,
    ...rest
  } = props;

  return (
    <div
      className={classNames(
        styles.backdrop,
        !showBackground && styles.noBackground,
        className
      )}
      {...rest}
    >
      {children}
    </div>
  );
};

Backdrop.displayName = "Backdrop";
