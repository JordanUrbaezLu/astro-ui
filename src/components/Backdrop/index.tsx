/* istanbul ignore file */
export * from "./Backdrop";
