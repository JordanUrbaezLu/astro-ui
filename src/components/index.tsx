export * from "./Alert";
export * from "./Button";
export * from "./Checkbox";
export * from "./DatePicker";
export * from "./Link";
export * from "./Menu";
export * from "./Tooltip";
export * from "./Select";
export * from "./Switch";
export * from "./TextField";
