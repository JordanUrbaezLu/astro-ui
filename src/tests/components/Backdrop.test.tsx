import { screen, render } from "@testing-library/react";
import { axe } from "jest-axe";
import { Backdrop } from "../../components/Backdrop";

describe("Backdrop", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<Backdrop />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(<Backdrop />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test("Should render with no background correctly.", () => {
    render(
      <Backdrop data-testid="Backdrop" showBackground={false} />
    );
    expect(screen.getByTestId("Backdrop")).toHaveClass(
      "noBackground"
    );
  });
});
