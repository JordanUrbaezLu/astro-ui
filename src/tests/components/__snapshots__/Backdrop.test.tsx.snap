// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop Should render component correctly. 1`] = `
<div
  class="backdrop"
/>
`;
